sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala

```diff
@@ -615,6 +620,7 @@ case class SortMergeJoinExec(
     }
 
     s"""
+       |$joinedRow = new JoinedRow();
```

Comment:
Hi, @rdblue .
Since this is a correctness issue, can we have a test case including the incorrect result you met?